content/shell/renderer/shell_content_renderer_client.cc

```diff
@@ -277,6 +283,73 @@ void ShellContentRendererClient::GetSupportedKeySystems(
 }
 #endif
 
+#if BUILDFLAG(IS_COBALT)
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+std::string GetMimeFromAudioType(const media::AudioType& type) {
+  // The MIME string is for very basic audio codec supportability check.
+  switch (type.codec) {
+    case media::AudioCodec::kAAC:
+      return "audio/mp4; codecs=\"mp4a.40.2\"";
+    case media::AudioCodec::kAC3:
+      return "audio/mp4; codecs=\"ac-3\"";
+    case media::AudioCodec::kEAC3:
+      return "audio/mp4; codecs=\"ec-3\"";
+    case media::AudioCodec::kOpus:
+      return "audio/webm; codecs=\"opus\"";
+    default:
+      return "";
+  }
+}
+
+std::string GetMimeFromVideoType(const media::VideoType& type) {
+  // The MIME string is for very basic video codec supportability check.
+  switch (type.codec) {
+    case media::VideoCodec::kH264:
+      return "video/mp4; codecs=\"avc1.4d4015\"";
+    case media::VideoCodec::kVP9:
+      return "video/webm; codecs=\"vp9\"";
+    case media::VideoCodec::kAV1:
+      return "video/mp4; codecs=\"av01.0.08M.08\"";
+    default:
+      return "";
+  }
+}
+
+bool ShellContentRendererClient::IsSupportedAudioType(
+    const media::AudioType& type) {
+  // TODO(b/376542844): Eliminate the usage of hardcoded MIME string once we
+  // support to query codec capabilities with configs.
+  if (type.codec != media::AudioCodec::kAAC &&
+      type.codec != media::AudioCodec::kAC3 &&
+      type.codec != media::AudioCodec::kEAC3 &&
+      type.codec != media::AudioCodec::kOpus) {
+    return false;
+  }
```

Comment:
nit: Instead of going through the supported codecs, we should remove the check here, and check whether the returned `mime` type is empty below.  This reduces maintenance cost as we only need to maintain one list of supported codecs.
If checking whether a string is empty is not elegant enough, we can always turn `GetMimeFromAudioType()` into `bool GetMimeFromAudioType(..., std::string* ...)`.